Signal/src/contact/SystemContactsFetcher.swift

```diff
@@ -434,15 +452,21 @@ class SystemContactsFetcher: NSObject {
 
         DispatchQueue.global().async {
 
-            var fetchedContacts: [Contact]?
+            guard self.tryToAcquireContactFetchLock() else {
```

Comment:
Seems like there's a race condition where we could lose important contact edits this way. If I launch the app, then while my contacts are still loading, I edit someone in my address book, I might not get that edit pulled into Signal until *next* contact pull.

Maybe that's an edge case that we can reasonably ignore, but we could also use a serial queue instead.

WDYT @charlesmchen

Reply:
Yup, that's the definite tradeoff here.  It's hard because this is a major perf win on iPhone 5C if you have a lot of contacts.

Maybe we can have two types of fetch requests: soft and hard.  Hard are always done, serially.  Soft can be de-duplicated.